PhysicsTools/PatUtils/interface/PATJetCorrExtractor.h

```diff
@@ -88,6 +69,24 @@ class PATJetCorrExtractor {
 
     return corrJetP4;
   }
+
+private:
+  static std::string format_vstring(const std::vector<std::string>& v) {
```

Comment:
This code is schoolbook example of memory-churn!  (how many times retVal is relocated?)
One may try this:
```
inline std::string format_vstring(const std::vector<std::string>& v)  {
    std::string retVal;
    auto ss = std::accumulate(v.begin(),v.end(),2,[](int a,std::string const & s){return a+s.length()+2;});
    retVal.reserve(ss);
    for_each(v.begin(),v.end(),[&](std::string const & s){retVal += (retVal.empty() ? "{ " : ", ")+s;});
    retVal+= " }";
    return retVal;
  }
```

Reply:
How about using fmt::format e..g `return "{ "+ fmt::format("{}",fmt::join(v,", ")) + " }";` should replace this whole function

Reply:
fmt is yet another library. When we move to c++20 we can use std::format
(btw when we move to c++20?)

Reply:
ok updated as suugested in https://github.com/cms-sw/cmssw/pull/34140#discussion_r655549971 .
Many of c++20 features are already available in GCC 10, I guess once we have gcc10 as production arch then we can enable c++20

Reply:
> When we move to c++20 we can use std::format

No we can't, unfortunately, because currently only MSVC implements it https://en.cppreference.com/w/cpp/compiler_support.